app/components/ChallengesMapView.js

```diff
@@ -71,17 +70,17 @@ export default class ChallengesMapView extends Component {
 				<View style={{flex:1}}>
 					<MapView
 						style={styles.map}
-						loadingEnabled={true}
-						showUserLocation={true}
+						loadingEnabled={false}
+						showsUserLocation={true}
 						initialRegion={this.props.initialRegion}
 					>
 					{this.props.primaryMarkers.map((marker, index) => (
-						<MapView.Marker 
+						<MapView.Marker
 							key={index}
 							identifier={marker.id}
 							coordinate={marker.latlng}
 							image={primaryMarkerImage}
-							onPress={this.onHeadMarkerPress}
+							onPress={() => this.onHeadMarkerPress(marker.id)}
```

Comment:
So, I made this change explicitly to pass in the `marker.id`. Why?

![screenshot_20170329-230429](https://cloud.githubusercontent.com/assets/5455419/24488663/ab210b50-14d5-11e7-8a07-480f01b2837e.png). It turns out that MainMapView:169 was throwing an error because challenge was undefined. So, I inspected the event getting passed in on line [30](https://github.com/dansteren/geo-challenge/pull/19/files#diff-493009816d47fbca7cf196afb91bd677R29) and it didn't have an `id` property on `e.nativeEvent`. So, I figured that we had to get the challenge somehow (it looked like that was the only real value you were using off of it far as I could tell) and this was the best way I could figure to do that. So, the changes here, and the changes down on [MainMapView:169](https://github.com/dansteren/geo-challenge/pull/19/files#diff-d18bf8947bd63088c10f0029280bd53cR169) were all just to fix that problem.

Maybe you're not getting this issue on iOS, but this was an issue I was experiencing on Android, so I figured I'd try to tackle it.

Let me know if you can't find a better way.